enhancements/insights/insights-runtime-extractor.md

```diff
@@ -0,0 +1,647 @@
+
+For example, inspecting the container allows to gather additional data such as:
+
+* The container (with the ID `sha256:5edd4b...`) is based on __RHEL 8.9__ ands runs a __Kafka Broker 3.6.1__ with __Java OpenJDK 17.0.9__
+* The container (with the ID `sha256:4redc...`)  is based on __Fedora 17__, and runs a __Quarkus 3.6.2__ application with __Java Eclipse Adoptium 21.0.1__ 
+* The container (with the ID `sha256:7euc...`)  is based on __RHEL 8.8__, and runs a __Red Hat Build of Quarkus 3.2.13__ application with __Red Hat Java 17.0.3-1__ 
+* The container (with the ID `sha256:2badc...`)  is based on __Alpine Linux v3.11__, and runs a __Spring Boot 2.7__ with __Java OpenJDK 17.0.8.1__
+* The container (with the ID `sha256:3fdb45...`) is based on __Ubuntu 20.04.4 LTS__, and runs a __Node.js v18.18.2__ application
+* The container (with the ID `sha256:5edd4b...`) is a scratch container (not layered on top of any image) and runs a __Go 1.18.10__ executable
+* The container (with the ID `sha256:566tr...`) is  based on __RHEL 8.9__ ands runs __PostgreSQL 10.23__
+
+This enhancement proposes to extract this additional layer of data to get deeper insights on the OpenShift workload.
+It proposes to do so while keeping the existing constraints and requirements that the Insights Operator already has for privacy, security & performance.
+
+## Motivation
+
+Currently, the Insights operator tries to identify what workload is used on OpenShift by looking for known fingerprinting in layers of containers, but there are a lot of issues with this approach where we can only identify base-level containers.
+We do, for example, not know if a python container is using AI libraries.
+Another issue is that we can only identify containers using known base containers (like OpenJDK), but we do not know what type of workload is more popular (Spring Boot, Quarkus, WildFly, EAP, etc.).
+
+The Insights operator can also not tell if a container is using a non-Red Hat provided base containers like Ubuntu or Debian. Knowing this data means that BU and product leadership can use this data to prioritize future work and realign investment.
+
+The proof of concept for this has been demonstrated to Red Hat leeadership, who agree that this is important data for us to know in order 
+to help shape the future of our product offerings.
+
+### User Stories
+
+* As a middleware product manager, I can observe the trends of a product version over time to estimate the expected value of any customer requested extended life offering.
+
+* As a product manager in the AI space, I can measure the application platforms/frameworks/languages used in clusters or namespaces running OpenShift AI to understand what users are coding applications in.
```

Comment:
so we could offer up-to-date images with those libraries and think how about how to support those better.